### DISCUSSION TITLE
Why is Prisma adding trailing zeros to decimal values in the database when a decimal does not define scale or precision?

### DISCUSSION THREAD

**Discussion by cbelsole** [prisma/prisma#20013] [Q&A]
### Bug description

[I created an example repo](https://github.com/cbelsole/prismatest) to show when I save `0.5` to the database it saves it as `0.5000000000000000`. I am not sure why it's not honoring the numbers original precision when none is defined on the schema.

### How to reproduce

Run the example repo.

### Expected behavior

When a decimal field is defined as `test_decimal Decimal @db.Decimal` in a schema, the original precision of the value should be honored since none is specified. This is the default behavior for Postgres when using raw SQL.

### Prisma information

```prisma
model DecimalTest {
  id           Int     @id @default(autoincrement())
  test_decimal Decimal @db.Decimal
}
```

```ts
import { PrismaClient } from "@prisma/client";
async function start() {
  const prisma = new PrismaClient();

  const testDecimal = await prisma.decimalTest.create({
    data: { test_decimal: 0.5 },
  });
  console.log(testDecimal);

  const testDecimals = await prisma.decimalTest.findMany();
  console.log(testDecimals);
}

start();

```

### Environment & setup

- OS: macOS
- Database: PostgreSQL
- Node.js version: v16.15.1

Example repo for env https://github.com/cbelsole/prismatest

### Prisma Version

```
4.16.1
```

**Comment by nurul3101** [comment]
Hello @cbelsole 👋 

I tried running the query that you shared, and I don't seem to get any trailing zeros.
I am using Prisma version 4.16.2. I am wondering if this could have something to do with your database configuration.

<img width="632" alt="Screenshot 2023-06-30 at 8 11 55 PM" src="https://github.com/prisma/prisma/assets/13049676/36fa0c76-1c13-4ab8-b3f3-72e4df24ee7f">

<img width="571" alt="Screenshot 2023-06-30 at 8 12 30 PM" src="https://github.com/prisma/prisma/assets/13049676/77878d09-f111-42b1-bf1b-d1fdc33b1ab1">

**Reply by nurul3101** [reply]
@cbelsole I am checking to see if you had a chance to look at my last response.

It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by cbelsole** [comment]
Hey @nurul3101. Thanks for getting back to me. I'm wondering how you're looking at this data. I reran the repo with a fresh database and when I queried the DB directly via `psql` I got:

```sql
postgres=# \d
                 List of relations
 Schema |        Name        |   Type   |  Owner   
--------+--------------------+----------+----------
 public | DecimalTest        | table    | postgres
 public | DecimalTest_id_seq | sequence | postgres
 public | _prisma_migrations | table    | postgres
(3 rows)

postgres=# select * from "DecimalTest";
 id |    test_decimal    
----+--------------------
  1 | 0.5000000000000000
(1 row)
```

This seems to be right in JS since when I log the Prisma output it shows:
```js
╰─$ npm start

> @cbelsole/prismatest@0.1.0 start
> tsc && node build/index.js

{ id: 1, test_decimal: 0.5 }
[ { id: 1, test_decimal: 0.5 } ]
```

But when you query the DB directly you get trailing zeros.

**Reply by nurul3101** [reply]
I was looking at the data through prisma studio by invoking the command `npx prisma studio`. I'll check by querying the data through `psql`.

**Reply by nurul3101** [reply]
@cbelsole I tried to query it through raw sql and I seem to get the correct response.
Which PostgreSQL version are you using?

<img width="713" alt="Screenshot 2023-07-04 at 6 09 00 PM" src="https://github.com/prisma/prisma/assets/13049676/5b86adcc-fdc5-4caf-a09e-115f79a364b7">

**Reply by cbelsole** [reply]
I'm running this against [postgres 14.1](https://github.com/cbelsole/prismatest/blob/main/docker-compose.yaml#L5).

**Reply by nurul3101** [reply]
I tested this with the same PostgreSQL version but can't seem to reproduce this. I believe the trailing zeros shouldn't affect the end result so it shouldn't cause an issue. Do you have any thoughts on what could be causing this?

**Reply by nurul3101** [reply]
Just checking to see if you are still running into this issue?

**Reply by nurul3101** [reply]
@cbelsole Please feel free to let us know in case you figured out what was causing the trailing zeros. I believe the trailing zeros shouldn't affect the end result but it would be good to figure out what was causing them in the first place. It might be helpful to others.

**Reply by nurul3101** [reply]
I'll mark this discussion as closed, please feel free to reply if needed.
